core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -530,33 +623,35 @@ boolean pointToValue(OpData opData) {
         // the straight forward helping didn't work, check why
         Operation operation = opData.op;
 
-        // the operation is remove, means we tried to change the handle index we knew about to -1
-        // the old handle index is no longer there so we have nothing to do. Note that in case of non-ZC removal the
+        // the operation is remove, means we tried to change the value reference we knew about to DELETED_VALUE
+        // the old value reference is no longer there so we have nothing to do. Note that in case of non-ZC removal the
         // loop in remove() will lookup the key again so we will still return the previous value
         if (operation == Operation.REMOVE) {
             return true; // this is a remove, no need to try again and return doesn't matter
         }
 
         // the operation is either NO_OP, PUT, PUT_IF_ABSENT, COMPUTE
-        int expectedHandleIdx = opData.handleIndex;
-        int foundHandleIdx = getEntryField(opData.entryIndex, OFFSET_HANDLE_INDEX);
+        long expectedValueReference = opData.newValueReference;
```

Comment:
Why expectedValueReference is getting newValueReference? Most likely it is just bad naming of variables... However, opData has old and new valueReferences so the expected should be old... Please check and explain in a code comment

Reply:
Previously there were `handleIndex` and `prevHandleIndex`. There is a condition checking whether someone else did the CAS for us when comparing `expectedValueReference` and `foundValueReference`.
To avoid naming problems I inlined this variable, and now the condition is `if (opData.newValueReference == foundValueReference)`.